### PR TITLE
mighty-mike: init at 3.0.2-unstable

### PR DESCRIPTION
**Pull request by NateEag**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/na/nanosaur/package.nix
Comment (yunfachi) [inline pkgs/by-name/na/nanosaur/package.nix:1]: Avoid modifying this file, as it is not related to the current commit

File: pkgs/games/bugdom/default.nix
Comment (yunfachi) [inline pkgs/games/bugdom/default.nix:1]: This one too

File: pkgs/by-name/mi/mighty-mike/package.nix
Comment (yunfachi) [inline pkgs/by-name/mi/mighty-mike/package.nix:53]:
```suggestion
  meta = {
    description = "Port of Mighty Mike, a 1995 Macintosh game by Pangea Software, for modern operating systems";
    longDescription = ''
      This is Pangea Software's Mighty Mike updated to run on modern systems.
      Set in a toy store, this top-down action game is a staple of 90's Macintosh games.
      It was initially published in 1995 under the name Power Pete.
    '';
    homepage = "https://jorio.itch.io/mightymike";
    license = lib.licenses.cc-by-sa-40;
    mainProgram = "MightyMike";
    platforms = lib.platforms.linux;
```

File: pkgs/by-name/mi/mighty-mike/package.nix
Comment (yunfachi) [inline pkgs/by-name/mi/mighty-mike/package.nix:13]:
```suggestion
  version = "3.0.2-unstable-2024-04-01";
```
Should match the commit date, not the pull request date

**Review by yunfachi — COMMENTED** [review]
You should add yourself to https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix in a new commit with the message `maintainers: add <your_handle>`.
It is preferable that this commit is older than the one where you add package.nix.
Also, add `maintainers = with maintainers; [your_handle];` to the `meta` attribute in package.nix.

File: pkgs/by-name/na/nanosaur/package.nix
Comment (NateEag) [reply pkgs/by-name/na/nanosaur/package.nix:1]: Whoops, I hadn't meant to commit these - was going to issue a separate PR for them

File: pkgs/by-name/mi/mighty-mike/package.nix
Comment (yunfachi) [inline pkgs/by-name/mi/mighty-mike/package.nix:51]:
```suggestion
    license = lib.licenses.cc-by-nc-sa-40;
```
NonCommercial https://github.com/jorio/MightyMike?tab=License-1-ov-file#readme

**Review by yunfachi — CHANGES_REQUESTED** [review]
Also, I think it’s better to change the commit message to `mighty-mike: init at 3.0.2-unstable-2024-04-01` (full version)

File: pkgs/by-name/mi/mighty-mike/package.nix
Comment (yunfachi) [inline pkgs/by-name/mi/mighty-mike/package.nix:52]:
```suggestion
    maintainers = with lib.maintainers = [ nateeag ];
    mainProgram = "MightyMike";
```

File: pkgs/by-name/mi/mighty-mike/package.nix
Comment (NateEag) [reply pkgs/by-name/mi/mighty-mike/package.nix:51]: Good catch, thanks!

File: pkgs/by-name/mi/mighty-mike/package.nix
Comment (NateEag) [reply pkgs/by-name/mi/mighty-mike/package.nix:52]: Oops, thanks - this is what I get for hacking too late at night. 😅

**Comment by NateEag** [comment]
@yunfachi This should be ready for re-review, if you're available.

File: pkgs/by-name/mi/mighty-mike/package.nix
Comment (yunfachi) [inline pkgs/by-name/mi/mighty-mike/package.nix:38]:
```suggestion
    runHook preInstall

    mkdir -p "$out/share/MightyMike"
    mv Data ReadMe.txt "$out/share/MightyMike/"

    install -Dm755 {.,$out/bin}/MightyMike
    wrapProgram $out/bin/MightyMike --chdir "$out/share/MightyMike"

    install -Dm644 $src/packaging/io.jor.mightymike.desktop $out/share/applications/mightymike.desktop
    install -Dm644 $src/packaging/io.jor.mightymike.png $out/share/pixmaps/mightymike-desktopicon.png

    runHook postInstall
```

**Review by yunfachi — APPROVED** [review]

**Comment by NateEag** [comment]
@yunfachi What's the next step in getting this merged?

I see the `passthru.tests` were skipped, apart from `aarch64-darwin` that's in `Queued` state. 

Is there something I should be doing about that?

**Comment by yunfachi** [comment]
> @yunfachi What's the next step in getting this merged?
> 
> I see the `passthru.tests` were skipped, apart from `aarch64-darwin` that's in `Queued` state.
> 
> Is there something I should be doing about that?

You can request a review from a nixpkgs committer, post in https://discourse.nixos.org/t/prs-ready-for-review/3032, and wait for the merge - it usually takes a lot of time.

**Comment by NateEag** [comment]
@yunfachi got it, thanks!

File: pkgs/by-name/mi/mighty-mike/package.nix
Comment (AndersonTorres) [inline pkgs/by-name/mi/mighty-mike/package.nix:27]:
```suggestion
  nativeBuildInputs = [
    SDL2
    cmake
    makeWrapper
  ];

  buildInputs = [ SDL2 ];

  strictDeps = true;
```

Comment (NateEag) [reply pkgs/by-name/mi/mighty-mike/package.nix]: @AndersonTorres Thanks! I've applied these changes and re-tested the build locally.

**Review by AndersonTorres — APPROVED** [review]

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**
